components/dropdown/dropdown-content-mixin.js

```diff
@@ -1001,6 +1009,14 @@ export const DropdownContentMixin = superclass => class extends LocalizeCoreElem
 				return spaceAround.right * -1;
 			}
 		}
+		if (this.align === 'start' || this.align === 'end') {
+			const shift = Math.min((targetWidth / 2) - (20 + 16 / 2), 0);
```

Comment:
The 20 corresponds approximately to 1rem, and the 16 is the pointer size. I'll add a comment to note this.
This is the same logic that tooltip uses for positioning.